libs/platform/ebpf_pinning_table.c

```diff
@@ -204,17 +207,22 @@ ebpf_pinning_table_delete(ebpf_pinning_table_t* pinning_table, const ebpf_utf8_s
     if (return_value == EBPF_SUCCESS) {
         entry = *existing_pinning_entry;
         return_value = ebpf_hash_table_delete(pinning_table->hash_table, (const uint8_t*)&existing_key);
+        // If unable to remove the entry from the table, don't delete it.
+        if (return_value != EBPF_SUCCESS) {
```

Comment:
How can hash_table_delete fail unless the key is not found? And you are holding the pinning_table lock here.

Reply:
The hash table implements RCU semantics. This means that on the deletion of a key it allocates a new block to hold the pointers to hashtable buckets. The allocation of the new block can fail, which causes the table update to fail.